source/robot_model/src/QPSolver.cpp

```diff
@@ -0,0 +1,157 @@
+#include "robot_model/QPSolver.hpp"
+
+namespace robot_model {
+
+QPSolver::QPSolver(
+    const unsigned int nb_joints, pinocchio::ModelTpl<double>::ConfigVectorType lower_position_limit,
+    pinocchio::ModelTpl<double>::ConfigVectorType upper_position_limit,
+    pinocchio::ModelTpl<double>::TangentVectorType velocity_limit)
+    : nb_joints_(nb_joints),
+      lower_position_limit_(lower_position_limit),
+      upper_position_limit_(upper_position_limit),
+      velocity_limit_(velocity_limit) {
+  // Initialize the QP solver
+  this->init_solver();
+}
+
+QPSolver::QPSolver(const QPSolver& other)
+    : nb_joints_(other.nb_joints_),
+      lower_position_limit_(other.lower_position_limit_),
+      upper_position_limit_(other.upper_position_limit_),
+      velocity_limit_(other.velocity_limit_),
+      hessian_(other.hessian_),
+      gradient_(other.gradient_),
+      constraint_matrix_(other.constraint_matrix_),
+      lower_bound_constraints_(other.lower_bound_constraints_),
+      upper_bound_constraints_(other.upper_bound_constraints_) {
+  this->init_solver();
+}
+
+bool QPSolver::init_solver() {
```

Comment:
This returns a bool but it's not used right? What would a false return value mean?

Reply:
A false return values mean that the solver is not initialized (failed to initialize). This is from `bool initSolver();` from `OsqpEigen` library and since we are wrapping around it, I decided to keep it the same as the previous interface.